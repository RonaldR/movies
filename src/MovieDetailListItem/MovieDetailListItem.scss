.list-item {
	display: flex;
	margin-bottom: 8px;
}

.list-item__label {
	flex-basis: 90px;
}

.list-item__item {
	font-weight: bold;
	flex: 1 0 0;
}
