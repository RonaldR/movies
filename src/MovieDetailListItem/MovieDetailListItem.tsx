import React from 'react';

import './MovieDetailListItem.scss';

type Props = {
    label: string,
    item: string,
    link?: string
}

const MovieDetailListItem = ({ label, item, link }: Props) => (
    <li className="list-item">
        <span className="list-item__label">{label}</span>
        <span className="list-item__item">
            {link
                ? <a href={link} target="_blank" rel="noopener noreferrer">{item}</a>
                : item}
        </span>
    </li>
);

export default MovieDetailListItem;