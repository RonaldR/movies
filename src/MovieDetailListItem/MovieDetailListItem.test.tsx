import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import MovieDetailListItem from './MovieDetailListItem';

test('movie detail list item', async () => {
    const { asFragment } = render(<MovieDetailListItem item="yes" label="Best movie ever" />);
    expect(asFragment()).toMatchSnapshot();
});
