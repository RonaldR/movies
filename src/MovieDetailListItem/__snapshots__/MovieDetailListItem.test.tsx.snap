// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movie detail list item 1`] = `
<DocumentFragment>
  <li
    class="list-item"
  >
    <span
      class="list-item__label"
    >
      Best movie ever
    </span>
    <span
      class="list-item__item"
    >
      yes
    </span>
  </li>
</DocumentFragment>
`;
